### PR TITLE
Add documentation how to render PlantUML in code blocks.

### PR DESCRIPTION
**Pull request by davidsvantesson**
I managed to get plantuml working by using a plantuml server. I think this would be quite similar how Gitlab does it. This PR suggest adding documentation of it to https://docs.gitea.io/en-us/customizing-gitea/.

Related: #3902.
Related #9553.

### CODE REVIEW

File: docs/content/doc/advanced/external-renderers.en-us.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ it is just a matter of:
 * add some configuration to your `app.ini` file
 * restart your Gitea instance
 
+This support rendering of complete files. If you wan't to render code blocks in markdown you would need to do something with javascript, see some examples on the [Customizing Gitea](../customizing-gitea) page.
```
Comment (guillep2k) [inline docs/content/doc/advanced/external-renderers.en-us.md]:
```suggestion
This supports rendering of whole files. If you wan't to render code blocks in markdown you would need to do something with javascript, see some examples on the [Customizing Gitea](../customizing-gitea) page.
```

The second correction might not be necessary, but sounds better to me.

File: docs/content/doc/advanced/customizing-gitea.en-us.md [hunk 2]
```diff
@@ -139,6 +139,39 @@ you would need to remove `{{if .RequireHighlightJS}}` and `{{end}}`.
 
 Mermaid will detect and use tags with `class="language-mermaid"`.
 
+#### Example: PlantUML
+
+You can add [PlantUML](https://plantuml.com/) support to Gitea's markdown by using a PlantUML server.
+The data is encoded and sent to the PlantUML server which generates the picture. There is an online 
+demo server at http://www.plantuml.com/plantuml, but if you (or your users) have sensitive data you 
+can set up your own [PlantUML server](https://plantuml.com/server) instead. To set up PlantUML rendering
```
Comment (guillep2k) [inline docs/content/doc/advanced/customizing-gitea.en-us.md]:
```suggestion
can set up your own [PlantUML server](https://plantuml.com/server) instead. To set up PlantUML rendering,
```

File: docs/content/doc/advanced/customizing-gitea.en-us.md [hunk 3]
```diff
@@ -139,6 +139,39 @@ you would need to remove `{{if .RequireHighlightJS}}` and `{{end}}`.
...
+copy javascript files from https://gitea.com/davidsvantesson/plantuml-code-highlight and put in your
```
Comment (guillep2k) [inline docs/content/doc/advanced/customizing-gitea.en-us.md]:
```suggestion
copy javascript files from https://gitea.com/davidsvantesson/plantuml-code-highlight and put them in your
```

File: docs/content/doc/advanced/customizing-gitea.en-us.md [hunk 4]
```diff
@@ -139,6 +139,39 @@ you would need to remove `{{if .RequireHighlightJS}}` and `{{end}}`.
...
+`custom/public` folder. Then add the following to `custom/footer.tmpl`:
+
```
Comment (guillep2k) [inline docs/content/doc/advanced/customizing-gitea.en-us.md:151]:
```suggestion

For the actual location of the `custom` directory, check the "Configuration" tab in the "Site Administration" page.

```

I think we should start introducing this kind of hints in the docs when possible. This is only a suggestion, of course.

Comment (davidsvantesson) [reply docs/content/doc/advanced/customizing-gitea.en-us.md:151]: I added information on top of the page instead. People have to learn reading the full page 😁

File: docs/content/doc/advanced/customizing-gitea.en-us.md [hunk 5]
```diff
@@ -18,7 +18,8 @@ menu:
 Customizing Gitea is typically done using the `CustomPath` folder - by default this is
 the `custom` folder from the running directory, but may be different if your build has
 set this differently. This is the central place to override configuration settings,
-templates, etc. You can check the `CustomPath` using `gitea help`. You can override
+templates, etc. You can check the `CustomPath` using `gitea help`. You can also find
+the path on the _Configuration_ tab in the _Site Administrator_ page. You can override
```
Comment (guillep2k) [inline docs/content/doc/advanced/customizing-gitea.en-us.md]:
```suggestion
the path on the _Configuration_ tab in the _Site Administration_ page. You can override
```

At least in the English locale.

File: docs/content/doc/advanced/customizing-gitea.en-us.md [hunk 6]
```diff
@@ -139,6 +140,39 @@ you would need to remove `{{if .RequireHighlightJS}}` and `{{end}}`.
...
+<script src="https://your-server.com/encode.js"></script>
+<script src="https://your-server.com/plantuml_codeblock_parse.js"></script>
+<script>
+<!-- Replace call with address to your plantuml server-->
+parsePlantumlCodeBlocks("http://www.plantuml..com/plantuml")
+</script>
+{{end}}
+```
+
+You can then add blocks like below to your markdown:
```
Comment (guillep2k) [inline docs/content/doc/advanced/customizing-gitea.en-us.md]:
```suggestion
You can then add blocks like the following to your markdown:
```

I'm not sure if the current text is proper English (not native myself).

File: docs/content/doc/advanced/external-renderers.en-us.md [hunk 7]
```diff
@@ -22,6 +22,8 @@ it is just a matter of:
 * add some configuration to your `app.ini` file
 * restart your Gitea instance
 
+This supports rendering of whole files. If you want to render code blocks in markdown you would need to do something with javascript, see some examples on the [Customizing Gitea](../customizing-gitea) page.
```
Comment (guillep2k) [inline docs/content/doc/advanced/external-renderers.en-us.md]:
```suggestion
This supports rendering of whole files. If you want to render code blocks in markdown you would need to do something with javascript. See some examples on the [Customizing Gitea](../customizing-gitea) page.
```

Sorry, missed before.

**Comment by guillep2k** [comment]
Would you mind changing the casing of PlantUML in the title for the changelog?

**Review by sapk — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

**Pull request merged**